sdks/java/io/parquet/src/test/java/org/apache/beam/sdk/io/parquet/ParquetIOTest.java

```diff
@@ -84,11 +85,31 @@ public void testWriteAndRead() {
     PCollection<GenericRecord> readBack =
         readPipeline.apply(
             ParquetIO.read(SCHEMA).from(temporaryFolder.getRoot().getAbsolutePath() + "/*"));
-
     PAssert.that(readBack).containsInAnyOrder(records);
     readPipeline.run().waitUntilFinish();
   }
 
+  @Test
+  public void testWriteAndReadWithSplit() {
+    List<GenericRecord> records = generateGenericRecords(1000);
```

Comment:
Please add a comprehensive set of unit tests for ParquetIO and BlockTracker splitting and progress tracking. You can use following for inspiration.
https://github.com/apache/beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/io/AvroSourceTest.java